### PR TITLE
Common tab component

### PR DESCRIPTION
**Pull request by Darsh-Tatvasoft**
Created CommonTab component with Storybook and tests
-Built CommonTab as a standalone, reusable component using MatTabGroup.
-Supports dynamic tabs with labels, optional icons, and safe HTML content rendering.
-Added Storybook stories showcasing various tab configurations for easy testing and reference.
-Wrote Jest unit tests covering rendering, label display, tab switching, and edge cases to ensure reliability.

-Testcases
{DD2005F0-D437-4253-8D75-7BE9A241815A}
{6838AFD7-46E1-4F62-A9F5-C32679EE9A3C}
{C01C0781-75CC-466C-932B-1DF5620EABF6}

-Tab Component examples
{EB7C9CB3-8D4F-4456-A462-55CB78B4C183}
{065A3A10-D0AE-4B7F-8BBF-A83F765DDCB3}

### CODE REVIEW

File: src/app/shared/components/tab-component/tab-component.component.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
+
+@Component({
+  selector: 'app-common-tab',
+  standalone: true,
+  imports: [MatTabsModule, MatIconModule, NgIf, NgFor],
+  templateUrl: './tab-component.component.html',
+  styleUrls: ['./tab-component.component.scss'],
+})
+export class TabComponentComponent {
```
Comment (hetvi-dhingani) [inline src/app/shared/components/tab-component/tab-component.component.ts]: rename this as TabComponent

File: src/app/shared/components/tab-component/tab-component.component.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  @Input() set tabs(value: { label: string; icon?: string; content?: string }[] | undefined) {
```
Comment (hetvi-dhingani) [inline src/app/shared/components/tab-component/tab-component.component.ts]: Create and use explicit TabInput/TabView types instead of inline type definitions

File: src/app/shared/components/tab-component/tab-component.component.ts [hunk 3]
```diff
@@ -11,24 +20,34 @@ import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
   templateUrl: './tab-component.component.html',
   styleUrls: ['./tab-component.component.scss'],
 })
-export class TabComponentComponent {
-  @Input() set tabs(value: { label: string; icon?: string; content?: string }[] | undefined) {
+export class TabComponent {
+  /** Dynamic tabs input: array of label, icon, and HTML content. Defaults to empty array if undefined. */
+  @Input() set tabs(value: TabInput[] | undefined) {
```
Comment (hetvi-dhingani) [inline src/app/shared/components/tab-component/tab-component.component.ts:25]: follow the consistent component structure as shown in shared document
